rj_param_utils/include/rj_param_utils/global_param.hpp

```diff
@@ -0,0 +1,31 @@
+/** @file */
+#pragma once
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/node.hpp>
+#include <rj_param_utils/param.hpp>
+#include <rj_param_utils/ros2_param_provider.hpp>
+
+
+namespace params {
+/**
+ * Implementation of a global parameter provider used by every node.
+ */
+
+class ROS2GlobalParamProvider : public ::params::ROS2ParamProvider {
+public:
+    explicit ROS2GlobalParamProvider(rclcpp::Node* node, const rclcpp::Node* global_node); 
```

Comment:
Last change on the skeleton: you don't need (and for some situations can't have) a pointer to the global parameter node, as it can be running in a separate binary. I would suggest just accepting a `const std::string&` for the node name (that's all the `AsyncParametersClient` needs for the remote node, anyway).